api/src/logic.ts

```diff
@@ -0,0 +1,29 @@
+import { ForestType, SiteConditionRule } from "./database/schema/forestType";
+
+export function recommend_plant(user_siteConditions: SiteConditionRule[], model_foresTypes: ForestType[]) {
```

Comment:
- SiteConditionRule just says what the acceptable values for site conditions are, the actual site condition itself should be of type `string`. 
- Use consistent naming convention, I'd prefer `camelCase` 
- Typo in `model_foresTypes`

Reply:
- Also use `let` instead of `var` if possible, its stricter